pyproject.toml

```
@@ -29,18 +29,18 @@ classifiers = [
 dynamic = ["version", "readme"]
 
 dependencies = [
-    "botocore >=1.34.70, <1.34.163", # NOTE: When updating, always keep `project.optional-dependencies` aligned
+    "botocore >=1.35.0, <1.35.4", # NOTE: When updating, always keep `project.optional-dependencies` aligned
```

Have you considered using the version compatibility operator here?
```suggestion
    "botocore ~= 1.35.0", # NOTE: When updating, always keep `project.optional-dependencies` aligned
```

It might simplify the maintenance (allowing compatibility with any last segment value).

---

This is a key safeguard for aiobotocore: Upstream introduces changes in patch versions that potentially break our integration. We need to manually check each upstream release and be diligent about setting this constraint.

There is a related section in https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore.